### PR TITLE
[#9141] Deprecate twisted.protocols.dict

### PR DESCRIPTION
**Pull request by hawkowl**
https://twistedmatrix.com/trac/ticket/9141

### CODE REVIEW

**Review by markrwilliams — APPROVED** [review]
@hawkowl correctly pointed out that this doesn't have meaningful tests.  If somebody wants to keep this, at the _very least_ it would need a lot of test.  Given that nobody wants to write those this is ready to land.

File: src/twisted/python/test/test_url.py [hunk 1]
```diff
@@ -802,11 +802,11 @@ def test_technicallyTextIsIterableBut(self):
 
 class URLDeprecationTests(SynchronousTestCase):
     """
-    L{twisted.python.constants} is deprecated.
+    L{twisted.python.url} is deprecated.
```
Comment (glyph) [inline src/twisted/python/test/test_url.py:805]: What is this change doing in this PR?

Comment (markrwilliams) [reply src/twisted/python/test/test_url.py:805]: I'm guessing @hawkowl couldn't help herself and cleaned up a mistake I introduced.

(I'm OK with this being in this PR)

Comment (hawkowl) [reply src/twisted/python/test/test_url.py:805]: yes

**Review by rodrigc — APPROVED** [review]

**Pull request merged**